src/main/flight/pid.c

```diff
@@ -803,77 +802,92 @@ void FAST_CODE pidController(const pidProfile_t *pidProfile, const rollAndPitchT
         }
 #endif // USE_YAW_SPIN_RECOVERY
 
-#if defined(USE_ABSOLUTE_CONTROL)
-        // mix in a correction for accrued attitude error
-        float acCorrection = 0;
-        if (acGain > 0) {
-            acCorrection = constrainf(axisError[axis] * acGain, -acLimit, acLimit);
-            currentPidSetpoint += acCorrection;
-        }   
-#endif
-        
         // -----calculate error rate
         const float gyroRate = gyro.gyroADCf[axis]; // Process variable from gyro output in deg/sec
-        float errorRate = currentPidSetpoint - gyroRate; // r - y
-
-        handleCrashRecovery(
-            pidProfile->crash_recovery, angleTrim, axis, currentTimeUs, gyroRate,
-            &currentPidSetpoint, &errorRate);
-
-        // --------low-level gyro-based PID based on 2DOF PID controller. ----------
-        // 2-DOF PID controller with optional filter on derivative term.
-        // b = 1 and only c (dtermSetpointWeight) can be tuned (amount derivative on measurement or error).
-
-        // -----calculate P component and add Dynamic Part based on stick input
-        pidData[axis].P = pidCoefficient[axis].Kp * errorRate * tpaFactor;
-        if (axis == FD_YAW) {
-            pidData[axis].P = ptermYawLowpassApplyFn((filter_t *) &ptermYawLowpass, pidData[axis].P);
-        }
 
-        // -----calculate I component
-        float itermErrorRate = 0.0f;
-#if defined(USE_ABSOLUTE_CONTROL)
+#ifdef USE_ABSOLUTE_CONTROL
+        float acCorrection = 0;
         float acErrorRate;
-#endif
+#endif        
+        float itermErrorRate = 0.0f;
+
 #if defined(USE_ITERM_RELAX)
         if (itermRelax && (axis < FD_YAW || itermRelax == ITERM_RELAX_RPY )) {
-            float gyroTargetLow = pt1FilterApply(&windupLpf[axis][0], currentPidSetpoint);
-            float gyroTargetHigh = pt1FilterApply(&windupLpf[axis][1], currentPidSetpoint);
-#if defined(USE_ABSOLUTE_CONTROL)
-            gyroTargetLow += acCorrection;
-            gyroTargetHigh += acCorrection;
-#endif
-            if (axis < FD_YAW) {
-                int itemOffset = (axis << 1);
-                DEBUG_SET(DEBUG_ITERM_RELAX, itemOffset++, gyroTargetHigh);
-                DEBUG_SET(DEBUG_ITERM_RELAX, itemOffset, gyroTargetLow);
+            const float gyroTarget = pt1FilterApply(&windupLpf[axis], currentPidSetpoint);
+            const float gyroHpf = fabsf(currentPidSetpoint - gyroTarget);
+            if (itermRelaxType == ITERM_RELAX_SETPOINT) {
+                itermErrorRate = (1 - MIN(1, fabsf(gyroHpf) / 60.0f)) * (currentPidSetpoint - gyroRate);
```

Comment:
gyroHpf is already absolute value ... 

Also it may be easier to read as
```
if(gyroHpf < 60) {
  itermErrorRate = (1 - gyroHpf / 60) * (currentPidSetpoint - gyroRate);
}

Reply:
However we want itermErrorRate to be 0 above 60.
I guess it could be:
```
if(gyroHpf < 60) {
    itermErrorRate = (1 - gyroHpf / 60) * (currentPidSetpoint - gyroRate);
} else
    itermErrorRate = 0
}
```

Reply:
The HPF value is already defined as an absolute `const float gyroHpf = fabsf(currentPidSetpoint - gyroTarget)`

Reply:
OK!  Clever. 
The debug for my setpoint mode should be a bit different. I personally would like the debug's to carry different parameters, so I know exactly how much of the modifier is being applied.
And perhaps the code would be clearer if:
```gyroTarget``` was re-named ```setpointLpf```
```gyroHpf ```was re-named ```setpointHpf```
since that's what they actually are...  :-)
Then it would read:
```
const float setpointLpf = pt1FilterApply(&windupLpf[axis], currentPidSetpoint);
const float setpointHpf = fabsf(currentPidSetpoint - gyroTarget);
if (itermRelaxType == ITERM_RELAX_SETPOINT && setpointHpf < 60) {
    itermErrorRate = (1 - setpointHpf / 60) * (currentPidSetpoint - gyroRate);
    if (axis == FD_ROLL) {
        DEBUG_SET(DEBUG_ITERM_RELAX, 0, setpointLpf);
        DEBUG_SET(DEBUG_ITERM_RELAX, 2, setpointHpf);
        DEBUG_SET(DEBUG_ITERM_RELAX, 2, itermErrorRate);
    }
}
if (itermRelaxType == ITERM_RELAX_GYRO ) {
    const float tolerance = setpointHpf * 1.0f;
    itermErrorRate = fapplyDeadband(setpointLpf - gyroRate, tolerance);
    if (axis == FD_ROLL) {
        DEBUG_SET(DEBUG_ITERM_RELAX, 0, setpointLpf);
        DEBUG_SET(DEBUG_ITERM_RELAX, 1, setpointLpf + tolerance);
        DEBUG_SET(DEBUG_ITERM_RELAX, 2, setpointLpf - tolerance);
        DEBUG_SET(DEBUG_ITERM_RELAX, 3, axisError[axis] * 10);
    }
}
```